bake/compose_test.go

```diff
@@ -888,6 +878,39 @@ services:
 	require.Equal(t, map[string]*string{"TEST_VALUE": ptrstr("abc"), "FOO_VALUE": ptrstr("abc")}, c.Targets[0].Args)
 }
 
+func TestUnknownField(t *testing.T) {
+	tmpdir := t.TempDir()
+	dt := []byte(`
+services:
+  webapp:
+    bar: baz
+    build:
+     context: .
+
+foo:
+  - bar.baz
+`)
+
+	chdir(t, tmpdir)
+	_, err := ParseComposeFiles([]File{{Name: "compose.yml", Data: dt}})
+	require.NoError(t, err)
+}
+
+func TestUnknownBuildField(t *testing.T) {
+	tmpdir := t.TempDir()
+	dt := []byte(`
+services:
+  webapp:
+    build:
+     context: .
+     foo: bar
+`)
+
+	chdir(t, tmpdir)
+	_, err := ParseComposeFiles([]File{{Name: "compose.yml", Data: dt}})
+	require.Error(t, err)
```

Comment:
Can we check something specific about the error, eg. should contain the name of the unknown key.